posthog/warehouse/api/external_data_source.py

```diff
@@ -610,3 +613,63 @@ def wizard(self, request: Request, *arg: Any, **kwargs: Any):
             status=status.HTTP_200_OK,
             data={str(key): value.model_dump() for key, value in configs.items()},
         )
+
+    @action(methods=["GET"], detail=False)
+    def dwh_scene_stats(self, request: Request, *arg: Any, **kwargs: Any):
+        """
+        Returns aggregated statistics for the data warehouse scene including total rows processed.
+        Used by the frontend data warehouse scene to display usage information.
+        """
+        billing_interval = ""
+        rows_synced = 0
+        data_modeling_rows = 0
+
+        try:
+            billing_manager = BillingManager(get_cached_instance_license())
+            org_billing = billing_manager.get_billing(organization=self.team.organization)
+
+            if org_billing and org_billing.get("billing_period"):
+                billing_period = org_billing["billing_period"]
+                billing_period_start = parser.parse(billing_period["current_period_start"])
+                billing_period_end = parser.parse(billing_period["current_period_end"])
+                billing_interval = billing_period.get("interval", "month")
+
+                usage_summary = org_billing.get("usage_summary", {})
+                billing_tracked_rows = usage_summary.get("rows_synced", {}).get("usage", 0)
+
+                all_external_jobs = ExternalDataJob.objects.filter(
+                    team_id=self.team_id,
+                    created_at__gte=billing_period_start,
+                    created_at__lt=billing_period_end,
+                    billable=True,
+                )
+                total_db_rows = all_external_jobs.aggregate(total=Sum("rows_synced"))["total"] or 0
+
+                pending_billing_rows = max(0, total_db_rows - billing_tracked_rows)
+
+                rows_synced = billing_tracked_rows + pending_billing_rows
+
+                data_modeling_jobs = DataModelingJob.objects.filter(
+                    team_id=self.team_id,
+                    created_at__gte=billing_period_start,
+                    created_at__lt=billing_period_end,
+                )
+                data_modeling_rows = data_modeling_jobs.aggregate(total=Sum("rows_materialized"))["total"] or 0
+
+        except Exception as e:
+            logger.exception("Could not retrieve billing information", exc_info=e)
+
+        return Response(
+            status=status.HTTP_200_OK,
+            data={
+                "billingInterval": billing_interval,
+                "externalData": {
+                    "billingPeriodEnd": billing_period_end,
+                    "billingPeriodStart": billing_period_start,
+                    "dataModelingRows": data_modeling_rows,
+                    "trackedBillingRows": billing_tracked_rows,
+                    "pendingBillingRows": pending_billing_rows,
+                    "totalRows": rows_synced,
+                },
+            },
+        )
```

Comment:
If billing doesn't return a billing period, or if we enter the `except` block, this will be trying to read local vars that dont exist (e.g. `billing_period_end`) and will just error out